packages/react-kit/src/components/form/Upload/UploadedFile.tsx

```diff
@@ -1,68 +1,116 @@
 import { ImageSquare, X, FilePdf } from "phosphor-react";
 import React, { useCallback } from "react";
-import styled from "styled-components";
+import styled, { css } from "styled-components";
 import { bytesToSize } from "../../../lib/bytes/bytesToSize";
 import { theme } from "../../../theme";
 
 import { Grid } from "../../ui/Grid";
 import ThemedButton from "../../ui/ThemedButton";
 import { Typography } from "../../ui/Typography";
+import { FileUploadWrapper } from "../Field.styles";
+import { UploadProps } from "../types";
 const colors = theme.colors.light;
 
-const AttachmentContainer = styled.div<{ $isLeftAligned: boolean }>`
+const AttachmentContainer = styled.div<{
+  $isLeftAligned: boolean;
+  $isPdfFile?: boolean;
+}>`
   position: relative;
   display: flex;
   cursor: pointer;
   align-items: center;
-  padding: 1rem;
-  background-color: ${({ $isLeftAligned }) =>
-    $isLeftAligned ? "inherit" : colors.lightGrey};
-  ${({ $isLeftAligned }) =>
-    $isLeftAligned ? `border: 2px solid ${colors.white}` : ""};
+
   color: ${({ $isLeftAligned }) => ($isLeftAligned ? "inherit" : colors.black)};
-  margin-bottom: 0.3rem;
+
+  ${({ $isPdfFile, $isLeftAligned }) =>
+    $isPdfFile
+      ? css`
+          width: 100%;
+        `
+      : css`
+          padding: 1rem;
+          background-color: ${$isLeftAligned ? "inherit" : colors.lightGrey};
+          ${$isLeftAligned ? `border: 2px solid ${colors.white}` : ""};
+        `}
+
   svg:nth-of-type(2) {
     position: absolute;
     right: 1rem;
   }
 `;
 
-interface Props {
+interface Props extends Pick<UploadProps, "theme" | "disabled"> {
   fileName: string;
   fileSize: number;
   base64Content?: string;
   showSize: boolean;
   color: "grey" | "white";
+  errorMessage: unknown;
   isPdfOnly?: boolean;
   handleRemoveFile?: () => void;
+  handleChooseFile?: () => void;
+  style?: React.CSSProperties;
 }
+
 export default function UploadedFile({
   fileName,
   fileSize,
   color,
   base64Content,
   showSize,
   isPdfOnly,
-  handleRemoveFile
+  handleRemoveFile,
+  handleChooseFile,
+  theme,
+  disabled,
+  style,
+  errorMessage
 }: Props) {
   const FileContent = useCallback(() => {
     return (
-      <>
-        {isPdfOnly ? <FilePdf size={23} /> : <ImageSquare size={23} />}
-        <Typography fontSize="1rem" fontWeight="400">
-          &nbsp;&nbsp; {fileName}
-        </Typography>
+      <FileUploadWrapper
+        $isPdfOnly={isPdfOnly}
+        data-disabled={disabled}
+        onClick={() => {
+          if (!isPdfOnly) {
+            handleChooseFile?.();
+          }
```

Comment:
why this if and not just `handleChooseFile?.();`?

Reply:
Did i have to implement logic to edit specified click defined by clicked element?